### ISSUE TITLE
rule proposal: `prefer-each-else`

### ISSUE THREAD

**Issue by WarningImHack3r** [sveltejs/eslint-plugin-svelte#1555]
### Motivation

People generally don't know about the fact that you can add [an `{:else}` branch to an `{#each}` block](https://svelte.dev/docs/svelte/each#Else-blocks). Using them is more idiomatic, more concise, and indents the code less.

### Description

A rule that detects patterns like:
```svelte
{#if x.length[ > 0]}
  {#each x ...}
  {/each}
{:else}
{/if}
```
or the opposite:
```svelte
{#if x.length === 0} <!-- or `!x.length` -->
{:else}
  {#each x ...}
  {/each}
{/if}
```
And suggests replacing them with an `{:else}` in the `{#each}` block.

### Examples

```svelte
<!-- ✓ GOOD -->
{#each x ...}
{:else}
  x is empty
{/each}

<!-- ✗ BAD -->
{#if x.length}
  {#each x ...}
  {/each}
{:else}
  x is empty
{/if}

{#if x.length > 0}
  {#each x ...}
  {/each}
{:else}
  x is empty
{/if}

{#if x.length === 0}
  x is empty
{:else}
  {#each x ...}
  {/each}
{/if}

{#if !x.length}
  x is empty
{:else}
  {#each x ...}
  {/each}
{/if}
```
